uaclient/daemon/retry_auto_attach.py

```diff
@@ -0,0 +1,228 @@
+import datetime
+import logging
+import time
+from typing import Optional
+
+from uaclient import files, messages, system
+from uaclient.api import exceptions as api_exceptions
+from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import (
+    FullAutoAttachOptions,
+    full_auto_attach,
+)
+from uaclient.config import UAConfig
+from uaclient.daemon import AUTO_ATTACH_STATUS_MOTD_FILE
+from uaclient.data_types import (
+    DataObject,
+    Field,
+    IntDataValue,
+    StringDataValue,
+)
+
+LOG = logging.getLogger("pro.daemon.retry_auto_attach")
+
+RETRY_INTERVALS = [
+    900,  # 15m (T+15m)
+    900,  # 15m (T+30m)
+    1800,  # 30m (T+1h)
+    3600,  # 1h  (T+2h)
+    7200,  # 2h  (T+4h)
+    14400,  # 4h  (T+8h)
+    28800,  # 8h  (T+16h)
+    28800,  # 8h  (T+1d)
+    86400,  # 1d  (T+2d)
+    86400,  # 1d  (T+3d)
+    172800,  # 2d  (T+5d)
+    172800,  # 2d  (T+7d)
+    259200,  # 3d  (T+10d)
+    259200,  # 3d  (T+13d)
+    345600,  # 4d  (T+17d)
+    345600,  # 4d  (T+21d)
+    432000,  # 5d  (T+26d)
+    432000,  # 5d  (T+31d)
+]
+FLAG_FILE_PATH = "/run/ubuntu-advantage/flags/auto-attach-failed"
+
+RetryOptions = FullAutoAttachOptions
+
+
+class RetryState(DataObject):
+    fields = [
+        Field("interval_index", IntDataValue),
+        Field("failure_reason", StringDataValue, required=False),
+    ]
+
+    def __init__(
+        self,
+        interval_index: int,
+        failure_reason: Optional[str],
+    ):
+        self.interval_index = interval_index
+        self.failure_reason = failure_reason
+
+
+OPTIONS_FILE = files.DataObjectFile(
+    RetryOptions,
+    files.UAFile(
+        "retry-auto-attach-options.json",
+        private=False,
+    ),
+    files.DataObjectFileFormat.JSON,
+)
+STATE_FILE = files.DataObjectFile(
+    RetryState,
+    files.UAFile("retry-auto-attach-state.json", private=False),
```

Comment:
I wonder if this file should be private. My only concern is just that it is possible to manipulate the failure message here

Reply:
Doesn't hurt to make it private and can avoid things like that. I'll change it